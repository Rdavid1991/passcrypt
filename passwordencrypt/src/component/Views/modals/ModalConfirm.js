import React from 'react';
import { Button } from '../../common';

export const ModalConfirm = ({handleCancel,existEmail}) => {
    return (
        <div className="modal fade" id="confirm">
            <div className="modal-dialog" style={{ top: '25%' }}>
                <div className="modal-content">
                    <div className="modal-body">
                        <h3 className="text-center">
                            { existEmail? "El email existe"  : "Se registro satisfactoriamente"}
                        </h3>
                        <div className="d-flex justify-content-center">
                            <Button
                                title="OK"
                                className="btn btn-success"
                                dataDismiss="modal"
                                click={existEmail? ()=>{}  : handleCancel}
                            />
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
};
