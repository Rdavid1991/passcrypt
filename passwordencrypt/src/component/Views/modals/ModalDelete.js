import React from 'react';

// eslint-disable-next-line react/prop-types
export const ModalDelete = ({ handleServiceDelete }) => {
    return (
        <div id="modal-confirm" className="modal fade">
            <div className="modal-dialog" style={{ marginTop: '15%' }}>
                <div className="modal-content">
                    <div className="modal-body text-center">
                        <h2>Â¿Desea borrar?</h2>
                        <hr />
                        <div className="d-flex justify-content-around">
                            <button
                                data-target="#modal-confirm"
                                data-dismiss="modal"
                                className="btn btn-secondary w-25"
                            >
                                No
                            </button>
                            <button
                                data-target="#modal-confirm"
                                data-dismiss="modal"
                                className="btn btn-primary w-25"
                                onClick={handleServiceDelete}
                            >
                                Si
                            </button>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
};
