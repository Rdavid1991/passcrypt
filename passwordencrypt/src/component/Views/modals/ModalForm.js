/* eslint-disable react/prop-types */
import React, { useEffect, useState } from 'react';
import { getDisableButton } from '../../../helpers/helpers';
import { Button, Input, InputPassword } from '../../common';

export const ModalForm = ({ handleSaveService, handleInputChange, inputChange, htmlId }) => {

    const { user, service, password } = inputChange;
    const [disableButton, setDisableButton] = useState(false)

    useEffect(() => {
        setDisableButton(getDisableButton(service, user, password))
    }, [inputChange])

    return (
        <div className="modal fade" id={htmlId}>
            <div className="modal-dialog">
                <div className="modal-content">
                    <div className="modal-header">
                        <h5 className="modal-title" >Modal title</h5>
                        <button type="button" className="close" data-dismiss="modal" aria-label="Close">
                            <span aria-hidden="true">&times;</span>
                        </button>
                    </div>
                    <div className="modal-body">
                        <form
                            onSubmit={handleSaveService}
                        >
                            <div
                                className="form-group"
                            >
                                <label htmlFor="service">Servicio</label>
                                <Input
                                    type="text"
                                    value={service}
                                    name="service"
                                    change={handleInputChange}
                                    disableButton={disableButton}
                                    setDisableButton={setDisableButton}
                                />
                            </div>
                            <div className="form-group">
                                <label htmlFor="user">Usuario</label>
                                <Input
                                    type="text"
                                    value={user}
                                    name="user"
                                    change={handleInputChange}
                                    disableButton={disableButton}
                                    setDisableButton={setDisableButton}
                                />

                            </div>
                            <div className="form-group position-relative">
                                <label htmlFor="password">Contrase√±a</label>
                                <InputPassword
                                    value={password}
                                    name="password"
                                    change={handleInputChange}
                                />
                            </div>
                            <div className="modal-footer">
                                <Button
                                    dataDismiss="modal"
                                    type="button"
                                    className="btn btn-secondary"
                                    title="Cancelar"
                                />
                                <Button
                                    type="submit"
                                    className="btn btn-primary"
                                    title="Guardar"
                                    disabled={disableButton}
                                />
                            </div>
                        </form>
                    </div>
                </div>
            </div>
        </div>
    );
};
