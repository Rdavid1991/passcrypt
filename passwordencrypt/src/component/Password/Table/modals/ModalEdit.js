/* eslint-disable react/prop-types */
import React, { useEffect } from 'react';
import { AES } from 'crypto-js';
import { useCreateForm } from '../../../../hooks/useCreateForm';
import { ModalForm } from './ModalForm';
import { initEdit } from '../../../../helpers/helpers';

export const ModalEdit = ({ handleServiceEdit, services, itemCode }) => {

    const [inputChange, handleInputChange, setEditValue] = useCreateForm( initEdit( services,itemCode ));

    const { id, user, service, password } = inputChange;

    useEffect(() => {
        setEditValue( initEdit( services,itemCode ));
    }, [itemCode]);

    const handleSaveService = ( ) => {
        handleServiceEdit({
            id,
            user   ,
            service ,
            password
        });
    };

    return (
        <ModalForm
            handleSaveService={handleSaveService}
            handleInputChange={handleInputChange}
            inputChange={inputChange}
            htmlId={"modal-edit"}
        />
    );
};
