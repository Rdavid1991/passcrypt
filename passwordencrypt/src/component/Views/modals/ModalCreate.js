import React from 'react';
import { useCreateForm } from '../../hooks/useCreateForm';
import { ModalForm } from './ModalForm';

// eslint-disable-next-line react/prop-types
export const ModalCreate = ({ handleServiceAdd }) => {

    const [inputChange, handleInputChange, reset] = useCreateForm({
        service : '',
        user    : '',
        password: ''
    });

    const { user, service, password } = inputChange;

    const handleSaveService = () => {

        handleServiceAdd({
            id: new Date().getTime(),
            user,
            service,
            password,
        });

        reset();
    };

    return (
        <ModalForm
            handleSaveService={handleSaveService}
            handleInputChange={handleInputChange}
            inputChange={inputChange}
            htmlId={"modal-form"}
        />
    );
};
