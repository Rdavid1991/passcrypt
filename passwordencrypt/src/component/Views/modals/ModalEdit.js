/* eslint-disable react/prop-types */
import React, { useEffect } from 'react';
import { useCreateForm } from '../../hooks/useCreateForm';
import { ModalForm } from './ModalForm';
import { initEdit } from '../../../helpers/helpers';

//import PropType from 'prop-types';

export const ModalEdit = ({ handleServiceEdit, services, itemCode }) => {

    const [inputChange, handleInputChange, setEditValue] = useCreateForm( initEdit( services,itemCode ));

    const { id, user, service, password } = inputChange;

    useEffect(() => {
        setEditValue( initEdit( services,itemCode ));
    }, [itemCode]);



    const handleSaveService = ( e ) => {
        e.preventDefault();

        handleServiceEdit({
            id,
            user   ,
            service ,
            password
        });

        window.$( '#modal-edit' ).modal( 'hide' );
    };

    return (
        <ModalForm
            handleSaveService={handleSaveService}
            handleInputChange={handleInputChange}
            inputChange={inputChange}
            htmlId={"modal-edit"}
        />
    );
};

/* ModalEdit.prototype = {
    itemCode: PropType.string
} */
