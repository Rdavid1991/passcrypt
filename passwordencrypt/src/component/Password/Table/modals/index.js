import { ModalCreate } from './ModalCreate';
import { ModalDelete } from './ModalDelete';
import { ModalEdit } from './ModalEdit';

export {
    ModalCreate,
    ModalDelete,
    ModalEdit
};