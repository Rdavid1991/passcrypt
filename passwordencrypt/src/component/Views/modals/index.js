import { ModalCreate } from './ModalCreate';
import { ModalDelete } from './ModalDelete';
import { ModalEdit } from './ModalEdit';
import {GeneratePassword} from './ModalGeneratePwd';

export {
    ModalCreate,
    ModalDelete,
    ModalEdit,
    GeneratePassword
};