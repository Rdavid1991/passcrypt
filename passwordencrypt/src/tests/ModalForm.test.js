import '@testing-library/jest-dom'
import React from 'react';
import { shallow } from 'enzyme';
import { ModalForm } from '../component/Views/modals/ModalForm';

describe('Pruebas a formulario', () => {

    const inputChange = {
        user: "test",
        service:"test",
        password:"test"
    }

    const handleSaveService = jest.fn();

    let wrapper = shallow(<ModalForm inputChange={inputChange} handleSaveService={handleSaveService}/>)

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow(<ModalForm inputChange={inputChange} handleSaveService={handleSaveService}/>)
    });

    test('Debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('Debe bloquear el boton guardar ', () => {

        console.log(wrapper);

        wrapper.find('form').simulate('submit' , { preventDefault(){}});

        expect(handleSaveService).toBeCalled();
    })
    
    
})
