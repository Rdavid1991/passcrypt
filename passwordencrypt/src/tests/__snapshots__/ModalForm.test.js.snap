// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a formulario Debe mostrarse correctamente 1`] = `
<div
  className="modal fade"
>
  <div
    className="modal-dialog"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
        >
          Modal title
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            ×
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <form
          onSubmit={[MockFunction]}
        >
          <div
            className="form-group"
          >
            <label
              htmlFor="service"
            >
              Servicio
            </label>
            <Input
              disableButton={false}
              name="service"
              setDisableButton={[Function]}
              type="text"
              value="test"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="user"
            >
              Usuario
            </label>
            <Input
              disableButton={false}
              name="user"
              setDisableButton={[Function]}
              type="text"
              value="test"
            />
          </div>
          <div
            className="form-group position-relative"
          >
            <label
              htmlFor="password"
            >
              Contraseña
            </label>
            <InputPassword
              name="password"
              value="test"
            />
          </div>
          <div
            className="modal-footer"
          >
            <Button
              className="btn btn-secondary"
              dataDismiss="modal"
              title="Cancelar"
              type="button"
            />
            <Button
              className="btn btn-primary"
              disabled={false}
              title="Guardar"
              type="submit"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
